core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/reconfigure-auth-service-app.yml

```diff
@@ -1,47 +1,56 @@
 ---
-- name: Change keycloak stateful set to use {{ image_registry_address }}
+- name: "Change keycloak stateful set to use {{ image_registry_address }}"
+  run_once: true
+  environment:
+    KUBECONFIG: /home/{{ admin_user.name }}/.kube/config
   block:
     - name: upgrade-master | Get keycloak statefulset name
-      environment:
-        KUBECONFIG: /home/{{ admin_user.name }}/.kube/config
       shell: |-
-        kubectl get statefulsets.apps --all-namespaces -o=jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.spec.template.spec.containers[].image}{"\n"}{end}'|
+        kubectl get statefulsets.apps --all-namespaces -o=jsonpath={{ _jsonpath }} |
         grep -i keycloak |
         awk '{print $1}'
+      vars:
+        _jsonpath: >-
+          '{range .items[*]}{.metadata.name}{"\t"}{.spec.template.spec.containers[].image}{"\n"}{end}'
       changed_when: false
       register: keycloak_ss_name
-      args:
+      args: &args
         executable: /bin/bash
+
     - name: upgrade-master | Get keycloak namespace
-      environment:
-        KUBECONFIG: /home/{{ admin_user.name }}/.kube/config
       shell: |-
-        kubectl get statefulsets.apps --all-namespaces -o=jsonpath='{range .items[*]}{.metadata.namespace}{"\t"}{.spec.template.spec.containers[].image}{"\n"}{end}'|
+        kubectl get statefulsets.apps --all-namespaces -o=jsonpath={{ _jsonpath }} |
         grep -i keycloak |
         awk '{print $1}'
+      vars:
+        _jsonpath: >-
+          '{range .items[*]}{.metadata.namespace}{"\t"}{.spec.template.spec.containers[].image}{"\n"}{end}'
       changed_when: false
       register: keycloak_namespace
-      args:
-        executable: /bin/bash
+      args: *args
+
     - name: upgrade-master | Get keycloak image
-      environment:
-        KUBECONFIG: /home/{{ admin_user.name }}/.kube/config
       shell: |-
-        kubectl get statefulsets.apps {{ keycloak_ss_name.stdout }} -n {{ keycloak_namespace.stdout }} -o=jsonpath='{$.spec.template.spec.containers[:1].image}'
+        kubectl get statefulsets.apps {{ keycloak_ss_name.stdout }} -n {{ keycloak_namespace.stdout }} -o=jsonpath={{ _jsonpath }}
+      vars:
+        _jsonpath: >-
+          '{$.spec.template.spec.containers[:1].image}'
       changed_when: false
       register: keycloak_repository
-      args:
-        executable: /bin/bash
       when:
         - not keycloak_namespace.stdout == ""
+
     - name: upgrade-master | Patch keycloak to use {{ image_registry_address }}
```

Comment:
I think this task could be run in a loop (on `keycloak_info.stdout_lines`) since there can be more than one statefulset that uses keycloak image.